test/integration/high-level-commands-test.js

```diff
@@ -0,0 +1,132 @@
+const { expect } = require('chai');
+const parallel = require('mocha.parallel');
+const { run, KeySymbol, resetTempDirectory, getPathInTempDirectory, makeFolderInTempDirectory } = require('./test-utils');
+
+parallel('high level commands test', () => {
+    let cmd;
+
+    before(() => {
+        resetTempDirectory();
+    });
+
+    beforeEach(() => {
+        cmd = 'ask';
+    });
+
+    it('| should init new skill', async () => {
+        const folderName = 'new-skill';
+        makeFolderInTempDirectory(`${folderName}/skill-package`);
+        makeFolderInTempDirectory(`${folderName}/lambda`);
+
+        const cwd = getPathInTempDirectory(folderName);
+
+        const args = ['init'];
+        const inputs = [
+            { match: '? Skill Id' },
+            { match: '? Skill package path' },
+            { match: '? Lambda code path' },
+            { match: '? Use AWS CloudFormation', input: 'n' },
+            { match: '? Lambda runtime' },
+            { match: '? Lambda handler' },
+            { match: '? Does this look correct?' }
+        ];
+
+        const result = await run(cmd, args, { inputs, cwd });
+
+        expect(result).include('succeeded');
+    });
+
+    it('| should set up, deploy, clone hosted skill', async () => {
+        let folderName = 'hosted-skill';
+        // new
+        let args = ['new'];
+        let inputs = [
+            { match: '? Choose the programming language' },
+            { match: '? Choose a method to host your skill' },
+            { match: '? Please type in your skill name' },
+            { match: '? Please type in your folder name', input: folderName }
+        ];
+
+        let result = await run(cmd, args, { inputs });
+
+        expect(result).include('Hosted skill provisioning finished');
```

Comment:
Typically integration tests tests like a black box. It is preferred to avoid white box testing that involve behind the scene implementation details.

For example, POST to an API and then directly connecting to the database to verify that data was inserted - this would be whitebox testing. As long as status 200, it is ok.


.git/config and .ask/ask-states.json are implementation details - unit tests would be more appropriate. 
With the same approach we should test whole folder structure and every single file.